test/e2e/registration_service_test.go

```diff
@@ -39,15 +44,375 @@ func (s *registrationServiceTestSuite) TestLandingPageReachable() {
+		assert.Equal(s.T(), http.StatusUnauthorized, resp.StatusCode)
+
+		defer resp.Body.Close()
+
+		body, err := ioutil.ReadAll(resp.Body)
+		require.NoError(s.T(), err)
+		require.NotNil(s.T(), body)
+
+		mp := make(map[string]interface{})
+		err = json.Unmarshal([]byte(body), &mp)
+		require.NoError(s.T(), err)
+
+		// Check token error.
+		tokenErr := mp["error"]
+		require.IsType(s.T(), tokenErr, "")
+		require.Equal(s.T(), "token contains an invalid number of segments", tokenErr.(string))
+	})
+
+	s.Run("verify_post_signup_valid_token", func() {
+		// Call signup endpoint with an valid token.
+		req, err := http.NewRequest("POST", s.route+"/api/v1/signup", nil)
+		require.NoError(s.T(), err)
+		req.Header.Set("Authorization", "Bearer "+token0)
+		req.Header.Set("content-type", "application/json")
+		client := getClient()
+
+		resp, err := client.Do(req)
+		require.NoError(s.T(), err)
+
+		defer resp.Body.Close()
+
+		body, err := ioutil.ReadAll(resp.Body)
+		require.NoError(s.T(), err)
+		require.NotNil(s.T(), body)
+		assert.Equal(s.T(), http.StatusAccepted, resp.StatusCode)
+	})
+
+	s.Run("verify_get_signup_error_no_token", func() {
+		// Call signup endpoint without a token.
+		req, err := http.NewRequest("GET", s.route+"/api/v1/signup", nil)
+		require.NoError(s.T(), err)
+		req.Header.Set("content-type", "application/json")
+		client := getClient()
+
+		resp, err := client.Do(req)
+		require.NoError(s.T(), err)
+
+		// Retrieve unauthorized http status code.
+		assert.Equal(s.T(), http.StatusUnauthorized, resp.StatusCode)
+
+		defer resp.Body.Close()
+
+		body, err := ioutil.ReadAll(resp.Body)
+		require.NoError(s.T(), err)
+		require.NotNil(s.T(), body)
+
+		mp := make(map[string]interface{})
+		err = json.Unmarshal([]byte(body), &mp)
+		require.NoError(s.T(), err)
+
+		// Check token error.
+		tokenErr := mp["error"]
+		require.IsType(s.T(), tokenErr, "")
+		require.Equal(s.T(), "no token found", tokenErr.(string))
+	})
+	s.Run("verify_get_signup_error_unknown_auth_header", func() {
+		// Call signup endpoint with invalid authroization header.
+		req, err := http.NewRequest("GET", s.route+"/api/v1/signup", nil)
+		require.NoError(s.T(), err)
+		req.Header.Set("Authorization", "1223123123")
+		req.Header.Set("content-type", "application/json")
+		client := getClient()
+
+		resp, err := client.Do(req)
+		require.NoError(s.T(), err)
+
+		// Retrieve unauthorized http status code.
+		assert.Equal(s.T(), http.StatusUnauthorized, resp.StatusCode)
+
+		defer resp.Body.Close()
+
+		body, err := ioutil.ReadAll(resp.Body)
+		require.NoError(s.T(), err)
+		require.NotNil(s.T(), body)
+
+		mp := make(map[string]interface{})
+		err = json.Unmarshal([]byte(body), &mp)
+		require.NoError(s.T(), err)
+
+		// Check token error.
+		tokenErr := mp["error"]
+		require.IsType(s.T(), tokenErr, "")
+		require.Equal(s.T(), "found unknown authorization header:1223123123", tokenErr.(string))
+	})
+	s.Run("verify_get_signup_error_invalid_token", func() {
```

Comment:
same here, looks like a duplicate of https://github.com/codeready-toolchain/toolchain-e2e/pull/48/files#diff-7ba0205692bfe0d01766275d77b38cd0R219

Reply:
there are two endpoints - GET and POST for signup. i'm testing both to make sure they work

Reply:
same here, I got confused by the similar test names :/